src/webapp/src/components/ProgressBar.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div class="progressBar">
+    <div class="progressBar__label">{{ this.title }} : {{ this.status }}</div>
+    <progress class="progress is-primary" v-bind:value="this.value" max="100">
+      {{ value }}%
+    </progress>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import cookies from "js-cookie";
+
+export default {
+  name: "ProgressBar",
+  data: function () {
+    return {
+      value: null,
+      status: "Undefined",
+      call: setInterval(this.updateStatus, 500),
+    };
+  },
+  props: {
+    link: String,
+    title: String,
+  },
+  methods: {
+    updateStatus: function () {
+      console.log(process.env.VUE_APP_API_ADDR + this.link);
+      axios
+        .get(process.env.VUE_APP_API_ADDR + this.link, {
+          headers: {
+            Authorization: cookies.get("Authorization"),
+          },
+        })
+        .then((res) => {
+          this.status = res.data["status"];
+          switch (this.status) {
```

Comment:
Here is another proposition: the action have 0 -> N steps, and each time you complete an action you increment this counter.
It you have 3 steps and you completed the first on, it means you are a 33% of the gauge